opentelemetry-sdk/src/metrics/meter.rs

```diff
@@ -533,6 +550,25 @@ fn validate_instrument_config(name: &str, unit: &Option<Cow<'static, str>>) -> M
     validate_instrument_name(name).and_then(|_| validate_instrument_unit(unit))
 }
 
+fn validate_buckets(buckets: &[f64]) -> MetricResult<()> {
```

Comment:
We should check for `f64::NAN`, `f64::INFINITY`, and `f64::NEG_INFINITY` as well.